### PR TITLE
Dynamic key reload for remote keymanager

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

Implements all scenarios for the remote keymanager outlined in https://docs.google.com/document/d/1fLdUi6AHQ5-wYOy6DCzW5JlLlq3VuI275-KoyEZ2Pmw/edit#heading=h.m87ybthlvflh. Because the keymanager does not have the ability to notify about key changes in real time, we check for new keys on every slot.

**Which issues(s) does this PR fix?**

Part of #8572

**Other notes for review**

I am not too happy about the type assertion and if/else branching inside `waitForActivation`, but I couldn't find a better way of handling the remote keymanager. We can't use the `case <-accountsChangedChan` because there is no external trigger which notifies about key changes, as is the case with other keymanager types.

### CODE REVIEW

File: validator/client/wait_for_activation.go [hunk 1]
```diff
@@ -24,7 +25,7 @@ import (
 func (v *validator) WaitForActivation(ctx context.Context, accountsChangedChan chan [][48]byte) error {
 	// Monitor the key manager for updates.
 	if accountsChangedChan == nil {
-		accountsChangedChan = make(chan [][48]byte)
+		accountsChangedChan = make(chan [][48]byte, 1)
```
Comment (rkapka) [inline validator/client/wait_for_activation.go:28]: This change is required to prevent a deadlock.

File: validator/client/runner.go [hunk 2]
```diff
@@ -100,7 +100,7 @@ func run(ctx context.Context, v iface.Validator) {
 		handleAssignmentError(err, headSlot)
 	}
 
-	accountsChangedChan := make(chan [][48]byte)
+	accountsChangedChan := make(chan [][48]byte, 1)
```
Comment (rkapka) [inline validator/client/runner.go:103]: This change is required to prevent a deadlock.

File: validator/client/wait_for_activation.go [hunk 3]
```diff
@@ -87,47 +88,87 @@ func (v *validator) waitForActivation(ctx context.Context, accountsChangedChan <
...
-		default:
-			res, err := stream.Recv()
-			// If the stream is closed, we stop the loop.
-			if errors.Is(err, io.EOF) {
-				break
-			}
-			// If context is canceled we return from the function.
+
+	remoteKm, ok := v.keyManager.(remote.RemoteKeymanager)
+	if ok {
```
Comment (rkapka) [inline validator/client/wait_for_activation.go:93]: The diff here looks like a mess, but I basically moved all existing code into the `else` branch and implemented the `if` branch.

**Review by rauljordan — APPROVED** [review]
Looks good, only change that seems scary was the buffered channel change. However, it seems it should have been buffered in the first place

**Pull request merged**